api/src/main/java/org/openmrs/module/oauth2login/authscheme/OAuth2User.java

```diff
@@ -94,14 +99,32 @@ public User toOpenmrsUser(Properties props) {
 	 * @return The corresponding value from the JSON, an empty String if none is found.
 	 */
 	public static String get(String userInfoJson, String propertyKey, Properties props, String defaultValue) {
-		String propertyValue = props.getProperty(propertyKey);
+		String propertyValue = props.getProperty(propertyKey, null);
 		String res = defaultValue;
 		if (!StringUtils.isEmpty(propertyValue)) {
 			res = JsonPath.read(userInfoJson, "$." + propertyValue);
 		}
 		return res;
 	}
 	
+	/**
+	 * Return a roles list based on the OAuth 2 properties mappings.
+	 * 
+	 * @param props The mappings between the user info fields and the corresponding OpenMRS
+	 *            user/person properties.
+	 * @return The list of roles
+	 */
+	public List<String> getRoles(Properties props) {
+
+		String rolesName = get(userInfoJson, MAPPINGS_PFX + PROP_ROLES, props,
+				null);
+		if (rolesName != null) {
+			return Arrays.asList(rolesName.split(","));
+		} else {
+			return new ArrayList<>();
+		}
+	}
```

Comment:
It all looks good to me 👍 
I would probably just 1) use streams to crush this `getRoles` method into a one-liner, and 2) probably add a new **OAuth2UserTest.java** to test this method a little.

Reply:
Done